mypy/checkexpr.py

```diff
@@ -3870,14 +3873,25 @@ def visit_conditional_expr(self, e: ConditionalExpr, allow_none_return: bool = F
 
     def analyze_cond_branch(self, map: Optional[Dict[Expression, Type]],
                             node: Expression, context: Optional[Type],
-                            allow_none_return: bool = False) -> Type:
+                            allow_none_return: bool = False,
+                            is_else: bool = False) -> Type:
         with self.chk.binder.frame_context(can_skip=True, fall_through=0):
+            if map is not None:
+                self.chk.push_type_map(map)
+            if is_else and context is not None and isinstance(node, CallExpr):
+                # When calling a function on the else part,
+                # we can face a generic function with multiple type vars.
+                # When inferecing it, `context` might be used instead of real args.
+                # Usually, we don't want that.
+                # https://github.com/python/mypy/issues/11049
+                if not is_subtype(self.accept(node), context, ignore_type_params=True):
```

Comment:
I think that the `self.accept(node)` should be called within `with self.msg.disable_errors():`, since we don't want it to report any errors. The different context could plausibly change the reported errors, due to overloaded functions perhaps.

Reply:
> This seems like a promising approach.

Thanks! 😊 

> I think that the self.accept(node) should be called within with self.msg.disable_errors():

Good catch! I will fix it right now.

> I want to check this with our big internal repository before merging, however

Please, report any errors back for me to fix. (after the `with self.msg.disable_errors():` change)